File: sion_fasion_shop/ProyectoMVC/SionFashioWebApplicationMVC/SionFashioWebApplicationMVC/Models/metodos_de_pago.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SionFashioWebApplicationMVC.Models;

[Table("metodos_de_pago")]
public partial class metodos_de_pago
{
    [Key]
    public int id_metodo_pago { get; set; }

    [Required(ErrorMessage = "El método de pago es obligatorio.")]
    [StringLength(50, ErrorMessage = "El método de pago no puede exceder los 50 caracteres.")]
    public string metodo_pago { get; set; } = null!;

    public int? id_factura { get; set; }

    [ForeignKey("id_factura")]
    [InverseProperty("metodos_de_pagos")]
    public virtual factura? id_facturaNavigation { get; set; }
}

File: sion_fasion_shop/Entregables/Trimestre V/Sistema de inventario en Node.Js/Bck-End SionFashion-Shop/models/metodosDePago.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const metodosDePagoSchema = new Schema({
    // Este campo se genera automáticamente
    // El id_metodo_pago no necesita ser declarado explícitamente en el esquema
    metodo_pago: {
        type: String,
        required: [true, 'El método de pago es obligatorio.'],
        maxlength: [50, 'El método de pago no puede exceder los 50 caracteres.']
    },
    id_factura: {
        type: Schema.Types.ObjectId,
        ref: 'Factura', // Asegúrate de que el modelo "Factura" esté definido
        default: null // Permite que sea nulo si no hay una factura asociada
    }
});

// Crea el modelo MetodosDePago
const MetodosDePago = mongoose.model('MetodosDePago', metodosDePagoSchema);

module.exports = MetodosDePago;
